dDatabase/CDClientDatabase/CDClientTables/CDDeletionRestrictionsTable.cpp

```diff
@@ -0,0 +1,57 @@
+#include "CDDeletionRestrictionsTable.h"
+#include "GeneralUtils.h"
+#include "eDeletionRestrictionsCheckType.h"
+
+CDDeletionRestriction CDDeletionRestrictionsTable::Default = {
+	.id = 0,
+	.restricted = false,
+	.ids = {},
+	.checkType = eDeletionRestrictionsCheckType::MAX
+};
+
+void CDDeletionRestrictionsTable::LoadValuesFromDatabase() {
+
+	uint32_t size = 0;
+	auto tableSize = CDClientDatabase::ExecuteQuery("SELECT COUNT(*) FROM CurrencyTable");
+	while (!tableSize.eof()) {
+		size = tableSize.getIntField(0, 0);
+
+		tableSize.nextRow();
+	}
+
+	tableSize.finalize();
+
+	auto& entries = GetEntriesMutable();
+
+	auto tableData = CDClientDatabase::ExecuteQuery("SELECT * FROM DeletionRestrictions");
+	while (!tableData.eof()) {
+		CDDeletionRestriction entry;
+		entry.id = tableData.getIntField("id", -1);
+		if (entry.id == -1) continue;
+		entry.restricted = tableData.getIntField("restricted", -1);
+		const std::string raw_ids = tableData.getStringField("ids", "");
+		if (!raw_ids.empty()) {
+			for (const auto& idstr : GeneralUtils::SplitString(raw_ids, ',')) {
+				if (!idstr.empty()) {
+					const auto id = GeneralUtils::TryParse<int32_t>(idstr).value_or(-1);
+					if (id != -1) entry.ids.push_back(id);
+				} 
+			}
+		}
+		entry.checkType = static_cast<eDeletionRestrictionsCheckType>(tableData.getIntField("checkType", 6)); // MAX
```

Comment:
I would like it if this was the enum value instead of just the 6 and a comment as if we added our own restriction, we would need to remember to update this instead of just having it be updated for us

Reply:
makes sense! I didn't have the enum in this file when I wrote this part

Reply:
actually, can't use the enum since it's a class.
this is the behavior in the client, and I can't really see modding needing other restrictions than the ones listed, but maybe I'm not that imaginative lol

Reply:
Is a static cast not doable?

Reply:
ah true, just would look ugly :P